db_upload_repo/db_upload_repo/views.py

```diff
@@ -48,7 +100,254 @@ def handle_uploaded_file(f, project):
+    """Reads file in 32Kb chunks"""
+    while True:
+        data = file_obj.read(32768)
+        if not data:
+            break
+        yield data
+
+
+def download_file(request, file_path):
+    # make sure that file exists within current directory
+    if os.path.exists(file_path):
+        file_name = os.path.basename(file_path)
+        response = StreamingHttpResponse()
+        response['Content-Disposition'] = 'attachment; filename=%s' % file_name
+        # set the content-type by guessing from the filename
+        response['Content-Type'] = mimetypes.guess_type(file_name)[0]
+        # set the content-length to the file size
+        response['Content-Length'] = os.path.getsize(file_path)
+        file_obj = open(file_path)
+        response.streaming_content = read_file_chunkwise(file_obj)
+        return response
+    else:
+        raise Http404(_('File does not exist'))
+
+
+def latest_download_view(request, project, file_name):
+    try:
+        assert user_has_permission_for_project(request, project)
+    except AssertionError:
+        return redirect_home(request)
+    file_path = os.path.join(latest_db_path(project), file_name)
+    return download_file(request, file_path)
+
+
+def historical_download_view(request, project, file_name):
+    try:
+        assert user_has_permission_for_project(request, project)
+    except AssertionError:
+        return redirect_home(request)
+    file_path = os.path.join(historical_db_path(project), file_name)
+    return download_file(request, file_path)
+
+
+def check_report_task_view(request, task_id):
+    from .tasks import create_report
+    return JsonResponse({'ready': create_report.AsyncResult(task_id).ready()})
+
+
+def create_report_view(request, project):
+    from .tasks import create_report
+    create_report.delay(project)
+    return HttpResponseRedirect(
+        reverse('project_report', kwargs={'project': project}))
+
+
+class ReportView(TemplateView):
+
+    template_name = "report/report.html"
+
+    def get(self, request, *args, **kwargs):
+        try:
+            assert user_has_permission_for_project(request, kwargs['project'])
+        except AssertionError:
+            return redirect_home(request)
+        return super(ReportView, self).get(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        context = super(ReportView, self).get_context_data(**kwargs)
+        try:
+            project = Project.objects.get(project_code=kwargs['project'])
+        except Project.DoesNotExist:
+            raise Http404(_('Project does not exist'))
+        context['project'] = kwargs['project']
+        fields = [
+            'last_sync',
+            'facility_name',
+            'num_content_sessions',
+            'time_content_sessions',
+        ]
+        context['headers'] = [{
+            'name':
+            FacilitySummary._meta.get_field(field).verbose_name,
+            'header':
+            field,
```

Comment:
For dicts, more common style is having key and value on same line, from what I've seen?

Reply:
I wanted to comment on the same thing, I find this much more readable:

```
items.append({
    "name": content,
    "size": stats.st_size,
    "last_modified": datetime.fromtimestamp(stats.st_mtime),
    "url": url_generator(content),
    "folder": False,
})
```

Reply:
Will lint.